modules/apigee_edge_teams/src/Form/TeamContextSwitcherForm.php

```diff
@@ -134,10 +164,10 @@ public function buildForm(array $form, FormStateInterface $form_state, array $te
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
     /** @var \Drupal\apigee_edge_teams\Entity\TeamInterface $team */
-    if (($team_id = $form_state->getValue('team_id'))
-      && ($teams = $form_state->get('teams'))
-      && (isset($teams[$team_id]))
-      && ($team = $teams[$team_id])) {
+    if ($teams = $form_state->get('teams')) {
```

Comment:
Still these sanity checks are unnecessary. The submitted team id always exists and valid and form state storage always contains the `teams` because you passed that from the form builder.